internal/webhook/external/refresh.go

```diff
@@ -99,7 +123,43 @@ func refreshWarehouses(
 		w,
 		http.StatusOK,
 		map[string]string{
-			"msg": fmt.Sprintf("refreshed %d warehouse(s)", len(warehouses)),
+			"msg": fmt.Sprintf("refreshed %d warehouse(s)", len(toRefresh)),
 		},
 	)
 }
+
+func shouldRefresh(wh kargoapi.Warehouse, repoURL string, qualifiers []string) (bool, error) {
+	var shouldRefresh bool
+	for _, s := range wh.Spec.Subscriptions {
+		switch {
+		case s.Git != nil && git.NormalizeURL(s.Git.RepoURL) == repoURL:
+			selector, err := commit.NewSelector(*s.Git, nil)
+			if err != nil {
+				return false, fmt.Errorf("error creating commit selector for Git subscription %q: %w",
+					s.Git.RepoURL, err,
+				)
+			}
+			shouldRefresh = slices.ContainsFunc(qualifiers, selector.MatchesRef)
+		case s.Image != nil && image.NormalizeURL(s.Image.RepoURL) == repoURL:
+			selector, err := image.NewSelector(*s.Image, nil)
+			if err != nil {
+				return false, fmt.Errorf("error creating image selector for Image subscription %q: %w",
+					s.Image.RepoURL, err,
+				)
+			}
+			shouldRefresh = slices.ContainsFunc(qualifiers, selector.MatchesTag)
+		case s.Chart != nil && helm.NormalizeChartRepositoryURL(s.Chart.RepoURL) == repoURL:
+			selector, err := chart.NewSelector(*s.Chart, nil)
```

Comment:
@fuskovic when we talked offline... is this the spot where you were mentioning the selector couldn't be initialized if the repo URL didn't have a leading `oci://`?

Reply:
yeah exactly